### PR TITLE
engine: local target build and deployer [ch3472]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @dbentley, @jazzdan,

Please review the following commits I made in branch maiamcc/local-build-deployer:

3f22b7602515f0bcd80af610dc12061500303b4d (2019-09-17 15:49:54 -0400)
more tests

76d3ecea5c6cd458accefe3be24dae247305b688 (2019-09-17 15:28:30 -0400)
tests

cc48aa49855f85bc2e5c2ceb028b1dbc0fcebfc7 (2019-09-17 15:05:54 -0400)
i think this works?

53bb2b03dc862e785a7b1bd8bf9999f857a7c37e (2019-09-17 12:11:26 -0400)
skeleton

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/engine/local_target_build_and_deployer.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+func (bd *LocalTargetBuildAndDeployer) run(ctx context.Context, c model.Cmd) error {
+	if len(c.Argv) == 0 {
+		panic("LocalTargetBuildAndDeployer tried to run empty command " +
+			"(should have been caught by Validate() )")
+	}
+
+	writer := logger.Get(ctx).Writer(logger.InfoLvl)
+	cmd := exec.CommandContext(ctx, c.Argv[0], c.Argv[1:]...)
```
Comment (dbentley) [inline internal/engine/local_target_build_and_deployer.go]: Would be good to say what the command is before running it, and maybe a pipelinestate with timing info

File: internal/engine/local_target_build_and_deployer.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	cmd.Stdout = writer
+	cmd.Stderr = writer
+
+	err := cmd.Run()
+	if err != nil {
+		// TODO(maia): any point in checking if it's an ExitError,
```
Comment (dbentley) [inline internal/engine/local_target_build_and_deployer.go]: Probably worth writing the error to the log, too.

File: internal/engine/local_target_build_and_deployer.go (hunk 1 above)
Comment (maiamcc) [reply internal/engine/local_target_build_and_deployer.go]: good call -- in the interest of getting this working E2E i'll leave a TODO/make a ticket

File: internal/engine/local_target_build_and_deployer.go (hunk 2 above)
Comment (maiamcc) [reply internal/engine/local_target_build_and_deployer.go]: pretty sure the error will be written to the log as part of the build failure?

File: internal/engine/local_target_build_and_deployer.go (hunk 1 above)
Comment (dbentley) [reply internal/engine/local_target_build_and_deployer.go]: I think for debugging at least printing the Argv would be helpful.

**Pull request merged**